framework/assets/yii.validation.js

```diff
@@ -236,9 +236,15 @@ yii.validation = (function ($) {
             }
         },
 
-        trim: function ($form, attribute, options) {
+        trim: function ($form, attribute, options, value) {
             var $input = $form.find(attribute.input);
-            var value = $input.val();
+            var isCheckAble = $input.find('[type=radio]').is('[type=radio]') || $input.find('[type=checkbox]').is('[type=checkbox]');
+
+            if (!isCheckAble) {
```

Comment:
Negation isn't needed here. Will fix separately.

Reply:
Seems to understand what is at stake. I will now make a change

Reply:
I saw the fix